src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -49,7 +56,30 @@ public void Process(Exception exception, SentryEvent sentryEvent)
 
                 foreach (var key in sentryException.Data.Keys)
                 {
-                    sentryEvent.SetExtra($"Exception[{i}][{key}]", sentryException.Data[key]);
+                    if (key.StartsWith("sentry", StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (key.StartsWith(ExceptionDataTagKey, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var data = sentryException.Data[key] as string ?? "<invalid>";
+                            //Set the key after the ExceptionDataTagKey string.
+                            sentryEvent.SetTag(key.Substring(ExceptionDataTagKey.Length), data);
+                        }
+                        else if (key.StartsWith(ExceptionDataContextKey, StringComparison.OrdinalIgnoreCase))
+                        {
+                            //Set the key after the ExceptionDataTagKey string.
+                            _ = sentryEvent.Contexts.AddOrUpdate(key.Substring(ExceptionDataContextKey.Length),
```

Comment:
need to make sure the key isn't just `sentry:context:` or this would break.

Reply:
It doesn't break from what I saw (+ added a test case for that rn)